Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepFindBlocks.cs

```diff
@@ -35,16 +38,23 @@ public BlockStoreInnerStepFindBlocks(ILoggerFactory loggerFactory)
         {
             this.logger.LogTrace("()");
 
-            while (await ShouldStopFindingBlocks(context) == false)
+            var batchSize = BlockStoreInnerStepContext.DownloadStackThreshold - context.DownloadStack.Count;
```

Comment:
In ShouldStopFindingBlocks we check to see if the DownloadStack's count has exceeded the DownloadStackThreshold so why do we need a batchSize to check? I think it comes to the same thing?

Reply:
In your implementation the code will ask for x blocks and then move on to the read blocks and wait for it to be drained.

That introduces a bottle neck, ideally we want to continuously ask for blocks and store blocks.

Currently the code loops over the two steps together so the result was
Loop over:
- Ask one block
- Take one block (and add to a pending commit list if the list is above a threshold then push)
> this gives a continuation loop that asks and pushes

After you previous change
- ask for 10 blocks
Loop:
- Take one block (and add to a pending commit list if the list is above a threshold then push)

The issue is we wait for the 10 blocks to download before we ask for the next 10 blocks.
The desire is:
- Ask 10 blocks
- 5 blocks arrived 
- Ask for another 5 blocks

Reply:
OK thanks for clarifying @dangershony! I understand now what the issue was with my implementation.